libs/ui/radio-group/brain/src/lib/brn-radio.component.ts

```diff
@@ -58,215 +41,119 @@ export class BrnRadioChange {
-	}
-
-	// eslint-disable-next-line @typescript-eslint/no-empty-function
-	private _removeUniqueSelectionListener: () => void = () => {};
-
-	private _previousTabIndex: number | undefined;
+	/**
+	 * The unique ID for the radio button input. If none is supplied, it will be auto-generated.
+	 */
+	public readonly id = input(`brn-radio-${++BrnRadioComponent._nextUniqueId}`);
 
-	@ViewChild('input')
-	public _inputElement?: ElementRef<HTMLInputElement>;
+	public readonly ariaLabel = input<string | undefined>(undefined, { alias: 'aria-label' });
 
-	/** Focuses the radio button. */
-	focus(origin?: FocusOrigin): void {
-		if (!this._inputElement) return;
-		if (origin) {
-			this._focusMonitor.focusVia(this._inputElement, origin);
-		} else {
-			this._inputElement.nativeElement.focus();
-		}
-	}
+	public readonly ariaLabelledby = input<string | undefined>(undefined, { alias: 'aria-labelledby' });
 
-	_markForCheck() {
-		this._changeDetector.markForCheck();
-	}
+	public readonly ariaDescribedby = input<string | undefined>(undefined, { alias: 'aria-describedby' });
 
-	ngOnInit() {
-		if (this.radioGroup) {
-			// If the radio is inside a radio group, determine if it should be checked
-			this.checked = this.radioGroup.value === this._value;
+	/**
+	 * The value this radio button represents.
+	 */
+	public readonly value = input.required<T>();
 
-			if (this.checked) {
-				this.radioGroup.selected = this;
-			}
+	/**
+	 * Whether the radio button is required.
+	 */
+	public readonly required = input<boolean, BooleanInput>(false, {
+		transform: booleanAttribute,
+	});
 
-			// Copy name from parent radio group
-			this.name = this.radioGroup.name;
-		}
+	/**
+	 * Event emitted when the checked state of this radio button changes.
+	 */
+	public readonly change = output<BrnRadioChange<T>>();
 
-		this._removeUniqueSelectionListener = this._radioDispatcher.listen((id, name) => {
-			if (id !== this.id && name === this.name) {
-				this.checked = false;
-			}
-		});
-	}
+	protected readonly inputId = computed(() => `${this.id()}-input`);
 
-	ngDoCheck(): void {
-		this._updateTabIndex();
-	}
+	private readonly _inputElement = viewChild.required<ElementRef<HTMLInputElement>>('input');
 
-	ngAfterViewInit() {
-		this._updateTabIndex();
-		this._focusMonitor.monitor(this._elementRef, true).subscribe((focusOrigin) => {
-			if (!focusOrigin && this.radioGroup) {
-				this.radioGroup._touch();
-			}
-		});
+	constructor() {
+		this._focusMonitor.monitor(this._elementRef, true);
 	}
 
-	ngOnDestroy() {
+	ngOnDestroy(): void {
 		this._focusMonitor.stopMonitoring(this._elementRef);
-		this._removeUniqueSelectionListener();
 	}
 
 	/** Dispatch change event with current value. */
-	private _emitChangeEvent(): void {
-		this.change.emit(new BrnRadioChange(this, this._value));
+	private emitChangeEvent(): void {
+		this.change.emit(new BrnRadioChange(this, this.value()));
 	}
 
-	_onInputClick(event: Event) {
+	protected onInputClick(event: Event) {
```

Comment:
```suggestion
	protected onInputClick(event: Event): void {
```

Should we add the explicit return type here also?